src/api/java/com/minecolonies/api/util/BlockPosUtil.java

```diff
@@ -636,4 +640,84 @@ public static Rotation getRotationFromRotations(final int rotations)
                 return Rotation.NONE;
         }
     }
+
+    /**
+     * Returns the first air position near the given start. Advances vertically first then horizontally
+     *
+     * @param start     start position
+     * @param vRange    vertical search range
+     * @param hRange    horizontal search range
+     * @param predicate check predicate for the right block
+     * @return position or null
+     */
+    public static BlockPos findAround(final IBlockReader world, final BlockPos start, final int vRange, final int hRange, final Predicate<BlockState> predicate)
+    {
+        if (vRange < 1 && hRange < 1)
+        {
+            return null;
+        }
+
+        BlockPos temp;
+        int y = 0;
+        int y_offset = 1;
+
+        for (int i = 0; i < hRange + 2; i++)
+        {
+            for (int steps = 1; steps <= vRange; steps++)
+            {
+                // Start topleft of middle point
+                temp = start.add(-steps, y, -steps);
+
+                // X ->
+                for (int x = 0; x <= steps; x++)
```

Comment:
I'd make that methds.

Reply:
nah that is a hassle since I would need additional tuple return types just for a simple loop